tools/build-tools/src/typeValidator/memberDecomposition.ts

```diff
@@ -0,0 +1,272 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    CallSignatureDeclaration,
+    ClassDeclaration,
+    ConstructorDeclaration,
+    DiagnosticCategory,
+    GetAccessorDeclaration,
+    IndexSignatureDeclaration,
+    MethodDeclaration,
+    MethodSignature,
+    ModifierableNode,
+    Node,
+    Project,
+    PropertyDeclaration,
+    PropertySignature,
+    Scope,
+    SetAccessorDeclaration,
+    Type,
+    TypeChecker,
+} from "ts-morph";
+import { decomposeType, decomposeTypes, GenericsInfo, typeToString } from "./typeDecomposition";
+import { BreakingIncrement, IValidator, log } from "./validatorUtils";
+
+function mergeIntoSet<T>(into: Set<T>, from: Set<T>) {
+    from.forEach((v) => into.add(v));
+}
+
```

Comment:
Is this the same as the union of the two sets?

Reply:
yep.  did i miss a built-in way of doing this?

Reply:
I'm dumbfounded, but no, you didn't. [https://2ality.com/2015/01/es6-set-operations.html](https://2ality.com/2015/01/es6-set-operations.html)

Reply:
nit: maybe comment that this is what the operation's doing, or call it Union or UnionWith or something